File: 83. Remove Duplicates from Sorted List/js/main.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function (head) {
  let dummy = new ListNode(0);
  dummy.next = head;
  let current = dummy.next;

  while (current !== null && current.next !== null) {
    if (current.next.val === current.val) {
      // Skip the node with the value to remove
      current.next = current.next.next;
    } else {
      current = current.next; // Move to the next node
    }
  }

  return dummy.next;
};

File: 83. Remove Duplicates from Sorted List/python/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
from typing import Optional
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0)
        dummy.next = head
        current = dummy.next
        while current!= None and current.next != None:
            if current.next.val == current.val:
                current.next = current.next.next
            else:
                current = current.next

        return dummy.next